ddtrace/internal/telemetry/writer.py

```diff
@@ -0,0 +1,197 @@
+from typing import Dict
+from typing import List
+from typing import Optional
+
+from ...internal import forksafe
+from ...settings import _config as config
+from ..agent import get_connection
+from ..agent import get_trace_url
+from ..compat import get_connection_response
+from ..compat import httplib
+from ..compat import monotonic
+from ..encoding import JSONEncoderV2
+from ..logger import get_logger
+from ..periodic import PeriodicService
+from ..runtime import get_runtime_id
+from ..utils.formats import get_env
+from ..utils.time import StopWatch
+from .data import get_application
+from .data import get_host_info
+
+
+log = get_logger(__name__)
+
+
+def _get_interval_or_default():
+    return float(get_env("instrumentation_telemetry", "interval_seconds", default=60))
+
+
+class TelemetryWriter(PeriodicService):
+    """
+    Periodic service which sends Telemetry request payloads to the agent
+    """
+
+    ENDPOINT = "telemetry/proxy/api/v2/apmtelemetry"
```

Comment:
is v2 still correct in the url? in the payloads below you are using `v1`. this is confusing.

Reply:
Ya this is what caught me off guard too. v2 here refers to the events platform api (where the agent proxy routes payloads). We're still on v1 of telemetry api

Reply:
I'll need to pin httpretty to `0.9.7`. This is the only version that supports python 2.7 and supports accessing all sent requests (httpretty.latest_requests)